src/Features/CSharp/Portable/CodeRefactorings/PushMemberUp/AbstractMemberPusher.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CodeActions;
+using System.Linq;
+using Microsoft.CodeAnalysis.Formatting;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.PushMemberUp
+{
+    internal abstract class AbstractMemberPusher
+    {
+        protected string Title { get; }
+
+        protected INamedTypeSymbol TargetTypeSymbol { get; }
+
+        protected SyntaxNode TargetSyntaxNode { get; }
+
+        protected SemanticModel SemanticModel { get; }
+
+        protected SyntaxNode UserSelectedNode { get; }
+
+        protected Document ContextDocument { get; }
+
+        internal AbstractMemberPusher(
+            INamedTypeSymbol targetTypeSymbol,
+            SemanticModel semanticModel,
+            SyntaxNode userSelectedNode,
+            Document contextDocument)
+        {
+            Title = $"Add to {targetTypeSymbol.Name}";
+            TargetTypeSymbol = targetTypeSymbol;
+            TargetSyntaxNode = TargetTypeSymbol.DeclaringSyntaxReferences.First().GetSyntax();
```

Comment:
not sure if this is guaranteed to not be empty.  also, we have helpers that try to find the right 'part' of a potentially 'partial' type.  we shoudl use those.